### PR TITLE
Use paging for OpenSea requests to fetch more unique tokens

### PR DESCRIPTION
**Pull request by jinchung**
Currently, it is set with a 1 minute delay after receiving a txn before refetching from OpenSea. For the cases I did, it was able to get the correct update by the 1 minute. This delay could potentially be reduced.

The recurring 15 second interval has been removed.

This has been tested on sending an ETH transaction (will not refresh), an ERC20 that was sent or received (will refresh), and a trade execution (will not refresh). I decided against checking these new transactions against an existing list of ERC20s for now, as I can do these preliminary checks with just the transaction data itself.

I tested the addresses of users with a ton of NFTs on simulator. It works! However, there was one user who had close to 2000 Cryptokitties, and the app became unusable after opening the token family for Cryptokitties. Basically, the app becomes unusable until all the photos in the family were loaded. I created a separate issue for Wojtek to investigate preventing the loading of all the tokens in the family until they become close to on screen.

### CODE REVIEW

File: src/redux/data.js [hunk 1]
```diff
@@ -44,6 +44,8 @@ import {
 } from '../references';
 import { ethereumUtils, isLowerCaseMatch } from '../utils';
 import { addCashUpdatePurchases } from './addCash';
+/* eslint-disable-next-line import/no-cycle */
```
Comment (brunobar79) [inline src/redux/data.js:47]: Is there a way we can prevent having circular dependencies?   `src/redux/data` => `src/redux/uniqueTokens` => `src/redux/data`?

File: src/redux/uniqueTokens.js [hunk 2]
```diff
@@ -64,46 +64,56 @@ export const uniqueTokensRefreshState = () => async (dispatch, getState) => {
...
     }
+
     saveUniqueTokens(uniqueTokens, accountAddress, network);
-    dispatch({
-      payload: uniqueTokens,
-      type: UNIQUE_TOKENS_GET_UNIQUE_TOKENS_SUCCESS,
-    });
   } catch (error) {
     dispatch({ type: UNIQUE_TOKENS_GET_UNIQUE_TOKENS_FAILURE });
-    captureException(error);
```
Comment (brunobar79) [inline src/redux/uniqueTokens.js:91]: I'd still capture this. If opensea API starts failing often we would know.

File: src/parsers/transactions.js [hunk 3]
```diff
@@ -86,7 +86,21 @@ export const parseTransactions = (
         remainingTransactions
       )
     : concat(updatedPendingTransactions, parsedNewTransactions);
-  return uniqBy(updatedResults, txn => txn.hash);
+
+  const potentialNftTransaction = appended
+    ? find(parsedNewTransactions, txn => {
+        return (
+          !txn.protocol &&
```
Comment (brunobar79) [inline src/parsers/transactions.js:93]: is this what you meant by not looking at known ERC20s and looking at tx data instead?

Comment (jinchung) [reply src/parsers/transactions.js:93]: yes!

**Comment by jinchung** [comment]
> 👏 👏 👏 👏
> 
> Totally out of scope but if we render the images inside a flatlist we would get rid of the blank spot.
> ^^ CC: @mikedemarais

I'm gonna update my comment, but for the user who had 2k cryptokitties, the app became unusable after opening up the cryptokitty family. I reduced it down to 350 for now, but basically they need to wait until all the images are loaded before it can become usable again. Wojtek has an idea of how to delay the loading of images until they are about to be on screen.

File: src/redux/uniqueTokens.js (hunk 2 above)
Comment (jinchung) [reply src/redux/uniqueTokens.js:91]: Sure sounds good

File: src/redux/data.js (hunk 1 above)
Comment (jinchung) [reply src/redux/data.js:47]: Not sure, I need to think about this. Data needs uniqueTokens in order to refetch from OpenSea on new transactions. UniqueTokens needs Data in order to update assets that have unique tokens filtered out (bc zerion currently doesn't distinguish between a normal token and an erc721 yet).

**Pull request merged**